cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -101,41 +102,42 @@ public void addStateVariablesToCgmes() {
     private void addVoltagesForTopologicalNodes() {
         PropertyBags voltages = new PropertyBags();
         // add voltages for TpNodes existing in the Model
-        for (PropertyBag tn : cgmes.topologicalNodes()) {
-            Bus b = network.getBusBreakerView().getBus(tn.getId(CgmesNames.TOPOLOGICAL_NODE));
+        if (network.getProperty("CGMES_topology").equals("NODE_BREAKER")) {
+            // TODO we need to export SV file data for NodeBraker
+            LOG.warn("NodeBreaker view require further investigation to map correctly Topological Nodes");
+            return;
+        }
+        for (Bus b : network.getBusBreakerView().getBuses()) {
             PropertyBag p = new PropertyBag(SV_VOLTAGE_PROPERTIES);
-            if (b != null) {
-                p.put(CgmesNames.ANGLE, fs(b.getAngle()));
-                p.put(CgmesNames.VOLTAGE, fs(b.getV()));
-                p.put(CgmesNames.TOPOLOGICAL_NODE, tn.getId(CgmesNames.TOPOLOGICAL_NODE));
-                voltages.add(p);
-            }
+            p.put(CgmesNames.ANGLE, fs(b.getAngle()));
+            p.put(CgmesNames.VOLTAGE, fs(b.getV()));
+            p.put(CgmesNames.TOPOLOGICAL_NODE, b.getId());
+            voltages.add(p);
         }
-        cgmes.add(originalSVcontext, "SvVoltage", voltages);
+        cgmes.add(svContext, "SvVoltage", voltages);
     }
 
     private void addVoltagesForBoundaryNodes() {
         PropertyBags voltages = new PropertyBags();
         // add voltages for TpNodes existing in the Model's boundaries
-        boundaryNodesFromDanglingLines.forEach((line, bnode) -> {
+        for (DanglingLine dl : network.getDanglingLines()) {
             PropertyBag p = new PropertyBag(SV_VOLTAGE_PROPERTIES);
-            DanglingLine dl = network.getDanglingLine(line);
             Bus b = dl.getTerminal().getBusBreakerView().getBus();
```

Comment:
Dangling lines should already have voltage of xnode as properties (v, angle). In PR#1455 I added some code to check if we have these properties, then use them; if not, we could compute voltage from dangling line impedance and known network voltage.